arangod/Aql/Functions.cpp

```diff
@@ -563,7 +563,15 @@ bool parameterToTimePoint(ExpressionContext* expressionContext,
   AqlValue const& value = extractFunctionParameterValue(parameters, parameterIndex);
 
   if (value.isNumber()) {
-    tp = tp_sys_clock_ms(milliseconds(value.toInt64()));
+    int64_t v = value.toInt64();
+    if (ADB_UNLIKELY(v < -62167219200000 || v > 253402300799999)) {
```

Comment:
why is it illegal to work with dates BC

Reply:
I think so far we haven't artificially bounded the range for our date operations, so internally they depended on what the date library supports. Which seems to be platform-dependent.

Supporting dates BC and after 9999-99-99 makes date string handling just more complicated. If we can always agree on year taking 4 digits, it simplifies things a lot. Plus it makes things portable, because this date range can be represented portably, the int64_t values with dates can be converted to double losslessly etc. This is not given if we greatly increase the supported range of dates.

No matter what the actual boundaries are, I think we should define and enforce them, so the operations are reliable and portable.
WDYT?

Reply:
Dates before 1582 AD don't make much sense anyway because there was no Gregorian calendar. For your dinosaur age determination database you would probably not use ISO dates anyway as you can not even tell the year for sure. Years past 9999 don't seem useful either, and such dates as well as dates before 0 are more of an extension to the ISO 8601 standard (the data exchange parties need to agree on using the 7 char dates for every date string if they also have dates outside of the default range). The useful sorting properties of ISO date strings do not hold for negative years.

👍 for limiting AQL to 0..9999

Reply:
@Simran-B : yes, makes perfect sense! :smile: